### DISCUSSION TITLE
Question about routes

### DISCUSSION THREAD

**Discussion by hs41-18** [edmundhung/remix-guide#18] [General]
Hi Edmund, I was reading some of your code to understand how remix works in practice. Noticed this code in the index route

```
export let loader: LoaderFunction = () => {
	throw redirect('/discover');
};
```

Having not found `discover` a route, I was wondering which functionality of Remix is this. Could you please redirect me to the relevant remix doc or explain it to me?

**Comment by edmundhung** [comment]
Hi @samirhasanov, it will be caught by [~/routes/__layout/$guide.tsx](https://github.com/edmundhung/remix-guide/blob/main/app/routes/__layout/%24guide.tsx). 

This is called [Dynamic Route Parameters](https://remix.run/docs/en/v1/api/conventions#dynamic-route-parameters). As I don't have a route file for `discover`, i.e `discover.tsx`, Remix (react-router) match this instead and render its content.

**Reply by hs41-18** [reply]
Thanks, got it.
